File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.java
class Solution {

    public List<Integer> findDuplicates(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        ArrayList<Integer> arr = new ArrayList<>();
        for (int num : nums) {
            if (set.contains(num)) {
                arr.add(num);
            } else {
                set.add(num);
            }
        }
        return arr;
    }
}

File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDuplicates = function (nums) {
  const arr = [];
  const set = new Set();
  for (const i in nums) {
    if (set.has(nums[i])) {
      arr.push(nums[i]);
    } else {
      set.add(nums[i]);
    }
  }
  return arr;
};
